typing/jkind.mli

```diff
@@ -900,3 +895,15 @@ module Debug_printers : sig
     val t : Format.formatter -> 'd Const.t -> unit
   end
 end
+
+(* These aliases are here to make sure various modules don't depend on [jkind] -
+   they would create a cycle causing the build system to error otherwise.  They
+   will be removed in the PR that adds abstract kinds, and until then they
+   ensure that the dependency structure needed for that PR isn't broken. *)
+type temp_cycle_check_subst = Subst.t
+
+type temp_cycle_check_env = Env.t
+
+module type temp_cycle_check_datarepr = module type of Datarepr
+
+module type temp_cycle_check_predef = module type of Predef
```

Comment:
Here's two type equalities that we're not currently exposing in this mli but I think we should:
- `Types.Jkind_const.t` and `Jkind.Const.t`
- `Types.Jkind_const.Builtin.t` and `Jkind.Const.Builtin.t`
(I think this may be my original mistake and not a product of this PR.)